src/moscot/problems/base/_base_problem.py

```diff
@@ -474,6 +475,120 @@ def _create_marginals(
     def _estimate_marginals(self, adata: AnnData, *, source: bool, **kwargs: Any) -> ArrayLike:
         return np.ones((adata.n_obs,), dtype=float) / adata.n_obs
 
+    @d.dedent
+    def set_xy(
```

Comment:
I think considering the rather the potential unintuitive behaviour, I would __always__ validate the data. What's a use case where this is not true?

Reply:
If you do not have exactly the same cell names but you are certain that the order is correct. As by default validation is required, I think it is fair to keep it this way. Do you have a strong opinion?